qcodes/tests/test_autoloadable_channels.py

```diff
@@ -0,0 +1,282 @@
+            cls, parent: Instrument=None, **kwargs) -> dict:
+        """
+        Find the smallest channel number not yet occupied. An optional keyword
+        `greeting` is extracted from the kwargs. The default is "Hello"
+        """
+        channels_str = parent.channel_catalog()
+        existing_channels = [int(i) for i in channels_str.split(",")]
+
+        new_channel = 1
+        while new_channel in existing_channels:
+            new_channel += 1
+
+        new_kwargs = {
+            "name": f"channel{new_channel}",
+            "channel": new_channel,
+            "greeting": kwargs.get("greeting", "Hello")
+        }
+
+        return new_kwargs
+
+    def __init__(
+            self,
+            parent: Union[Instrument, 'InstrumentChannel'],
+            name: str,
+            channel: int,
+            greeting: str,
+            existence: bool = False,
+            channel_list: 'AutoLoadableChannelList' = None,
+    ) -> None:
+
+        super().__init__(parent, name, existence, channel_list)
+        self._channel = channel
+        self._greeting = greeting
+
+        self.add_parameter(
+            "hello",
+            get_cmd=f":INST:CHN{self._channel}:HLO"
+        )
+
+    def _create(self)->None:
+        """Create the channel on the instrument"""
+        self.parent.root_instrument.write(
+            f":INST:CHN:ADD {self._channel}, {self._greeting}")
+
+    def _remove(self)->None:
+        """Remove the channel from the instrument"""
+        self.write(f":INST:CHN:DEL {self._channel}")
+
+
+class DummyInstrument(Instrument):
+    """
+    This dummy instrument allows the creation and deletion of
+    channels
+    """
+
+    def __init__(self, name: str)->None:
+        super().__init__(name)
+
+        self._backend = MockBackend()
+
+        self.add_parameter(
+            "channel_catalog",
+            get_cmd=":INST:CHN:CAT",
+        )
+
+        channels = AutoLoadableChannelList(
+            self, "channels", SimpleTestChannel, channels_to_skip=["5"])
+        self.add_submodule("channels", channels)
+
+    def write_raw(self, cmd: str)->None:
+        self._backend.send(cmd)
+
+    def ask_raw(self, cmd: str)->str:
+        return self._backend.send(cmd)
+
+
+@pytest.fixture(scope='function')
+def dummy_instrument():
+    instrument = DummyInstrument("instrument")
+    yield instrument
+    instrument.close()
+
+
+def test_sanity(dummy_instrument):
+    """
+    Test the basic functionality of the auto-loadable channels, without using
+    the auto-loadable channels list
+    """
+    # Test that we are able to discover instruments automatically
+    channels = SimpleTestChannel.load_from_instrument(dummy_instrument)
+    assert len(channels) == 4
+    assert channels[0].hello() == "Hello from channel 1"
+    assert channels[1].hello() == "Hello from channel 2"
+    assert channels[2].hello() == "Hello from channel 4"
+    assert channels[3].hello() == "Hello from channel 5"
```

Comment:
why is this channel with `5` here if it was supposed to be skipped thanks to the `channels_to_skip=["5"]`? or am i misunderstanding something?

it would be great if there was an explicit assert/test for the `channels_to_skip=["5"]` part with some explanation in the comments (i appreciate the comments on this in SimpleTestChannel though).

Reply:
We the `channels_to_skip=["5"]` code is called when using the ChannelList. In this test, `load_from_instrument` is called without any kwargs. So all channels are present. 

I will make an additional test to see if the kwargs are propagated correctly.

Reply:
On second thought, I added the following to the docstring of the test: 

Please note that the `channels_to_skip` argument in the dummy instrument applies when accessing channels from the channels list. Since we are calling `load_from_instrument` directly in this test without this keyword argument, we will see all channels.